apps/api.planx.uk/modules/send/downloadApplicationFiles/index.ts

```diff
@@ -41,6 +47,13 @@ export async function downloadApplicationFiles(
       });
     }
 
+    // Get flow ID, in order to access flow submission email
+    const flowId = await getFlowId(sessionId);
+
+    // Get the flow submission email, which will run parallel to getTeamEmailSettings for now
+    const submissionEmail = await getFlowSubmissionEmail(flowId);
+    console.log(submissionEmail);
```

Comment:
So eventually this will swap out the clause at lines 32-39 here. We're currently checking the singular email from team settings, but we want to be checking the emails from `flow_integrations`.

Right now, these are synchronised, and there's no way to change them (break them).

It's probably worth just running this check now directly - it will have the same effect and we won't need to come back to fix this.

Reply:
This will also allow us to start writing tests immediately here...!

If we set up mock queries to handle the new queries added in this PR, we can then test the various scenarios here.

I'm not sure if we've talked through API tests before (if we have, it's certainly been a while). We can either catch up on a call, or I can leave you async notes on how to set this up.

Reply:
Here's the current tests for this endpoint for reference - https://github.com/theopensystemslab/planx-new/blob/oz/read-flow-submission-email/apps/api.planx.uk/modules/send/email/index.test.ts#L254-L326

You can see query mocks for the `GetTeamEmailSettings` table - what we'd be looking for here is setting up mocks to cover your queries.